docker/circleci/Dockerfile

```diff
@@ -0,0 +1,91 @@
+FROM circleci/golang:1.15-node-browsers
+
+# Install dependencies.
+RUN sudo apt-get update && \
+    sudo apt-get install -y --no-install-recommends \
+        build-essential \
+        bzr \
+        ca-certificates \
+        clang \
+        cmake \
+        curl \
+        gnupg \
+        libgmp-dev \
+        libmpc-dev \
+        libmpfr-dev \
+        libssl-dev \
+        libtinfo5 \
+        libtool \
+        libxml2-dev \
+        llvm-dev \
+        lzma-dev \
+        netcat-openbsd \
+        patch \
+        xutils-dev \
+        zlib1g-dev && \
+    sudo rm -rf /var/lib/apt/lists/*
+
+WORKDIR /tmp
+
+# Install native musl, and cross-compilers.
+ENV MUSL_VERSION=1.1.24
+RUN curl https://musl.libc.org/releases/musl-${MUSL_VERSION}.tar.gz -O && \
+    tar xzf musl-${MUSL_VERSION}.tar.gz && \
+    cd musl-${MUSL_VERSION} && \
+    ./configure && \
+    make && \
+    sudo make install && \
+    cd .. && \
+    rm -rf musl-${MUSL_VERSION}*
+# TODO: Convert these to curl calls to download from S3.
```

Comment:
Before merge: complete this TODO

Reply:
Update: if all we need is ARM64, we can revert to building the cross-compiler here. It'll still be slow, but not unbearably so.